code/modules/station_goals/secondary/supply/random_smithed_item.dm

```diff
@@ -0,0 +1,135 @@
+/datum/station_goal/secondary/random_smithed_item
+	name = "Random Smithed Item"
+	department = "Smith"
+	progress_type = /datum/secondary_goal_progress/random_smithed_item
+	weight = 10
+	var/obj/item/smithed_item/product_type
+	/// What quality should the product be?
+	var/datum/smith_quality/quality
+	/// What should the material type be?
+	var/datum/smith_material/material
+	/// What do you get for accomplishing this goal?
+	var/reward
+
+/datum/station_goal/secondary/random_smithed_item/randomize_params()
+	var/list/valid_types = list()
+	for(var/S in subtypesof(/obj/item/smithed_item))
+		var/obj/item/smithed_item/possible_item = S
```

Comment:
```suggestion
		var/obj/item/smithed_item/possible_item_type = S
```

Reply:
try and use `_type` for var suffixes when you're handling a type path, to make it easier for readers to follow along, and below